src/orchestration/KubernetesClient.js

```diff
@@ -54,8 +75,15 @@ class KubernetesClient {
       const ip = pod.status.podIP;
       const engine = { networks: [{ ip }], labels };
       const key = pod.metadata.uid;
+      const replicaSet = replicaMap.get(pod.metadata.ownerReferences[0].uid);
```

Comment:
So if it does not exist, does it become `undefined`?
There simply is no property in the response, or the key is is there but with value `undefined`?